engine/src/main/java/org/terasology/logic/console/commands/CoreCommands.java

```diff
@@ -253,10 +249,29 @@ public String search(@CommandParam("searched") String searched) {
      * @return List of blocks that match searched string
      */
     private List<String> findBlockMatches(String searchLowercase) {
-        return assetManager.getAvailableAssets(BlockFamilyDefinition.class)
-                .stream().<Optional<BlockFamilyDefinition>>map(urn -> assetManager.getAsset(urn, BlockFamilyDefinition.class))
-                .filter(def -> def.isPresent() && def.get().isLoadable() && matchesSearch(searchLowercase, def.get()))
-                .map(r -> new BlockUri(r.get().getUrn()).toString()).collect(Collectors.toList());
+        ResourceUrn curUrn;
+        List<String> l=new ArrayList<String>();
+        for (ResourceUrn urn : assetManager.getAvailableAssets(BlockFamilyDefinition.class)) {
+            // Current urn for logging purposes to find the broken urn
+            curUrn = urn;
+            try{
+                Optional<BlockFamilyDefinition> def = assetManager.getAsset(urn, BlockFamilyDefinition.class);
+                if (def.isPresent() && def.get().isLoadable() && matchesSearch(searchLowercase, def.get())) {
+                    String s = new BlockUri(def.get().getUrn()).toString();
```

Comment:
See my above comment about naming. (Although this variable could easily be joined with the following line